src/Database/Redis/Connection.hs

```diff
@@ -201,8 +204,27 @@ connectCluster bootstrapConnInfo = do
     case commandInfos of
         Left e -> throwIO $ ClusterConnectError e
         Right infos -> do
-            pool <- createPool (Cluster.connect infos shardMapVar Nothing) Cluster.disconnect 1 (connectMaxIdleTime bootstrapConnInfo) (connectMaxConnections bootstrapConnInfo)
+            let 
+                isConnectionReadOnly = connectReadOnly bootstrapConnInfo
+                clusterConnection = Cluster.connect infos shardMapVar Nothing isConnectionReadOnly
+            pool <- createPool (clusterConnect isConnectionReadOnly clusterConnection) Cluster.disconnect 1 (connectMaxIdleTime bootstrapConnInfo) (connectMaxConnections bootstrapConnInfo)
             return $ ClusteredConnection shardMapVar pool
+    where
+        clusterConnect :: Bool -> IO Cluster.Connection -> IO Cluster.Connection
+        clusterConnect readOnlyConnection connection = do
+            clusterConn@(Cluster.Connection nodeMap _ _ _ _) <- connection
+            nodesConns <-  sequence $ ( PP.fromCtx . (\(Cluster.NodeConnection ctx _ _) -> ctx ) . snd) <$> (HM.toList nodeMap)
+            void $ if readOnlyConnection
```

Comment:
remove `void` it's already returning `unit`. 
use `when` in place of `if with empty else`.

Reply:
done